File: RetrofitSample/app/src/main/java/com/unittest/retrofitsample/melisma/activity/MainActivity.java
package com.unittest.retrofitsample.melisma.activity;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.unittest.retrofitsample.CallRetrofit;
import com.unittest.retrofitsample.R;
import com.unittest.retrofitsample.melisma.callback.SearchResponse;
import com.unittest.retrofitsample.melisma.model.dto.LoginReqUserDto;
import com.unittest.retrofitsample.melisma.model.vo.MusicVo;
import com.unittest.retrofitsample.melisma.service.MusicService;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    List<String> urls;
    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        urls = new ArrayList<>();

        CallRetrofit retrofit = new CallRetrofit();
        MusicService musicService = new MusicService();

//        retrofit.login(new LoginReqUserDto("username", "password"));

        retrofit.login(new LoginReqUserDto("users", "password"));
//        retrofit.login(new Member("test@gmail.com", "password"));
        Button button = findViewById(R.id.search_btn);
        EditText codeEditText = findViewById(R.id.codeEditText);

        Button add_btn = findViewById(R.id.add_btn);
        add_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String code = codeEditText.getText().toString();
                if(code != "")
                    urls.add(code);
            }
        });
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
//                UUID id = UUID.randomUUID();
//                MusicDto model = new MusicDto(id, "url", 10, id);
//                musicService.createMusic(model);
                musicService.searchMusics(new SearchResponse() {
                    @Override
                    public void onSuccess(@NonNull List<MusicVo> vos) {
                        for (MusicVo vo: vos) {
                            Log.d("onSuccess search", vo.getId().toString());
                            urls.add(vo.getMusicUrl());
                        }
                    }

                    @Override
                    public void onError(@NonNull Throwable throwable) {

                    }
                });

//                String id = "74e052d2-561a-402d-a8b3-0b6261cb63f9";
//
//                musicService.searchMusic(UUID.fromString(id));
//                model.setViews(100);
//                model.setMusicUrl("musicUrl");
////                musicService.change(model);
////                retrofit.hello();
//                musicService.update(model);
            }
        });

        Button youtube_btn = findViewById(R.id.youtube_btn);

        youtube_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, YoutubeActivity.class);
                intent.putStringArrayListExtra("urls",  (ArrayList<String>)urls);
                startActivity(intent);
            }
        });
    }


}

File: KakaoAddressSearch/app/src/main/java/com/cyj/kakaoaddresssearch/MainActivity.kt
package com.cyj.kakaoaddresssearch

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    // 초기변수설정
    private lateinit var edit_addr: EditText

    // 주소 요청코드 상수 requestCode
    private val SEARCH_ADDRESS_ACTIVITY = 10000


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // UI 요소 연결
        edit_addr = findViewById(R.id.edit_addr)

        // 터치 안되게 막기
        edit_addr.setFocusable(false);
        // 주소입력창 클릭
        edit_addr.setOnClickListener(View.OnClickListener {
            Log.i("주소설정페이지", "주소입력창 클릭")
            val status = NetworkStatus.getConnectivityStatus(applicationContext)
            if (status == NetworkStatus.TYPE_MOBILE || status == NetworkStatus.TYPE_WIFI) {
                Log.i("주소설정페이지", "주소입력창 클릭")
                val i = Intent(applicationContext, AddressApiActivity::class.java)
                // 화면전환 애니메이션 없애기
                overridePendingTransition(0, 0)
                // 주소결과
                startActivityForResult(i, SEARCH_ADDRESS_ACTIVITY)
            } else {
                Toast.makeText(applicationContext, "인터넷 연결을 확인해주세요.", Toast.LENGTH_SHORT).show()
            }
        })
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
        super.onActivityResult(requestCode, resultCode, intent)
        Log.i("test", "onActivityResult")
        when (requestCode) {
            SEARCH_ADDRESS_ACTIVITY -> if (resultCode == RESULT_OK) {
                Log.i("result", "result in")

                val data = intent?.extras!!.getString("data")
                if (data != null) {
                    Log.i("test", "data:$data")
                    edit_addr!!.setText(data)
                }
            }
        }
    }
}